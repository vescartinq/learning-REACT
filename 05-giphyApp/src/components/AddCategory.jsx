import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState(''); // ''

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  // Funci√≥n que se ejecuta al clickar enter (<form onSubmit={handleSubmit}></form>)
  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((newInput) => [inputValue, ...newInput]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
