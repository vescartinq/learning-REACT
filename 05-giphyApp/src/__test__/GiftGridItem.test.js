import React from 'react';
// import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGridItem } from './../components/GifGridItem';

describe('GiftGridItem', () => {
  let wrapper;

  const title = 'Avengers';
  const url = 'https://localhost/algo.jpg';

  beforeEach(() => {
    //   Title y URL son required en PropTypes

    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test('should show GiftGridItem component working ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should use title received by props and show on paragraph', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('should use image received by props url and alt and show on img', () => {
    const img = wrapper.find('img');
    console.log(img.html()); //  <img src="https://localhost/algo.jpg" alt="Avengers"/>

    expect(img.props().src).toBe(url); //img recibe por props src y alt
    expect(img.props().alt).toBe(title); //img recibe por props src y alt
  });

  test('should have animate__fadeIn', () => {
    const div = wrapper.find('div');
    // console.log(div.props());
    //  {
    //   className: 'card animate__animated animate__fadeIn',
    //   children: [
    //     {
    //       '$$typeof': Symbol(react.element),
    //       type: 'img',
    //       key: null,
    //       ref: null,
    //       props: [Object],
    //       _owner: null,
    //       _store: {}
    //     },
    //     {
    //       '$$typeof': Symbol(react.element),
    //       type: 'p',
    //       key: null,
    //       ref: null,
    //       props: [Object],
    //       _owner: null,
    //       _store: {}
    //     }
    //   ]
    // }

    expect(div.hasClass('animate__fadeIn')).toBeTruthy();
  });
});
