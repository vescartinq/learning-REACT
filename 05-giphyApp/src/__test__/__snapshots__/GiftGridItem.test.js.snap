// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftGridItem should show GiftGridItem component working  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Avengers"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Avengers
     
  </p>
</div>
`;
